addons/purchase/models/purchase.py

```diff
@@ -1017,7 +1017,7 @@ def _prepare_purchase_order_line(self, product_id, product_qty, product_uom, val
 
         taxes = product_id.supplier_taxes_id
         fpos = po.fiscal_position_id
-        taxes_id = fpos.map_tax(taxes) if fpos else taxes
+        taxes_id = fpos.map_tax(taxes, self.product_id, seller.name) if fpos else taxes
```

Comment:
shouldn't it be the `product_id` given in parameter?

Reply:
True